rapidy/web_app.py

```diff
@@ -3,87 +3,78 @@
 import logging
 import warnings
 from functools import partial, update_wrapper
-from typing import Any, cast, Dict, Iterable, List, Mapping, Optional, Union
+from typing import Any, cast, Dict, Iterable, Iterator, List, Mapping, Optional, Tuple
 
 from aiohttp.abc import AbstractMatchInfo
-from aiohttp.helpers import AppKey
 from aiohttp.log import web_logger
-from aiohttp.web_app import (
-    _AppSignal,
-    _MiddlewaresHandlers,
-    _RespPrepareSignal,
-    _Subapps,
-    Application as AiohttpApplication,
-    CleanupContext,
-    CleanupError,
-)
+from aiohttp.web_app import Application as AiohttpApplication, CleanupError
 from aiohttp.web_exceptions import HTTPUnprocessableEntity
-from aiohttp.web_urldispatcher import MatchInfoError, UrlMappingMatchInfo
-from aiosignal import Signal
-from frozenlist import FrozenList
+from aiohttp.web_middlewares import _fix_request_current_app
+from aiohttp.web_urldispatcher import MatchInfoError
 
 from rapidy import hdrs
-from rapidy._annotation_container import AnnotationContainer, create_annotation_container
+from rapidy._annotation_container import AnnotationContainer, create_annotation_container, HandlerEnumType
 from rapidy._client_errors import _normalize_errors
 from rapidy.constants import CLIENT_MAX_SIZE
 from rapidy.media_types import ApplicationJSON
 from rapidy.typedefs import Handler, Middleware
 from rapidy.web_request import Request
 from rapidy.web_response import StreamResponse
+from rapidy.web_urldispatcher import UrlDispatcher
 
 __all__ = (
     'Application',
     'CleanupError',
 )
 
-from rapidy.web_urldispatcher import UrlDispatcher
-
 
 class Application(AiohttpApplication):
     def __init__(
             self,
             *,
             logger: logging.Logger = web_logger,
+            router: Optional[UrlDispatcher] = None,
             middlewares: Iterable[Middleware] = (),
```

Comment:
I think we should use some default value to have the possibility to empty a middleware list or use a default middleware list.

Reply:
can you give me an example?

Reply:
can you give me an example?